src/Umbraco.Core/Persistence/Repositories/VersionableRepositoryBase.cs

```diff
@@ -454,7 +454,6 @@ INNER JOIN
 
                             //this property has tags, so we need to extract them and for that we need the prevals which we've already looked up
                             var preValData = allPreValues.Value.Where(x => x.DataTypeNodeId == property.PropertyType.DataTypeDefinitionId)
-                                .Distinct()

```

Comment:
@Shazwazza makes sense, I'll look into this and do another pull request for the issue.